### DISCUSSION TITLE
posts/c-implement-a-mini-stackless-coroutine-framework-minico/

### DISCUSSION THREAD

**Discussion by giscus** [pluveto/blog-comments#5] [Announcements]
# posts/c-implement-a-mini-stackless-coroutine-framework-minico/

https://www.less-bug.com/posts/c-implement-a-mini-stackless-coroutine-framework-minico/

**Comment by eczn** [comment]
深度好文，我可以转载到我自己博客吗？

**Reply by pluveto** [reply]
欢迎转载，注明出处即可

**Comment by sptuan** [comment]
请教下，最后使用汇编保存程序栈桢的这种做法，实际上是向有栈协程发展了吧

**Reply by pluveto** [reply]
有栈的必要条件是**具备独立的栈空间**。这里的实现仅保存当前栈帧（只是为了保存和恢复局部变量），根本上依赖主栈，所以本质上达不到有栈协程的标准。

个人观点，欢迎讨论。
